regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +113,177 @@ def construct_global_ctx(self):
-            return
-        rc.index = rc.index.replace(" ", "")
-        if "-" in rc.index:
-            idx_parsed = [i for i in range(int(rc.index.split('-')[0]),
-                                           int(rc.index.split('-')[1]) + 1)]
-        elif "," in rc.index:
-            idx_parsed = [int(i) for i in rc.index.split(',')]
-        else:
-            idx_parsed = [int(rc.index)]
         new_mil = []
-        for idx in idx_parsed:
-            pdoc = {}
-            if idx == 1:
-                mil = {}
-                if not rc.due_date or not rc.name or not rc.objective:
-                    raise RuntimeError(
-                        "name, objective, and due date are required for a new milestone")
-                mil.update({'due_date': rc.due_date})
-                mil['due_date'] = get_due_date(mil)
-                mil.update({'objective': rc.objective, 'name': rc.name, 'uuid': get_uuid()})
-                if rc.audience:
-                    mil.update({'audience': rc.audience})
-                else:
-                    mil.update({'audience': ['lead', 'pi', 'group_members']})
-                if rc.type:
-                    mil.update({'type': rc.type})
-                else:
-                    mil.update({'type': 'meeting'})
-                if rc.status:
-                    mil.update({'status': rc.status})
-                else:
-                    mil.update({'status': 'proposed'})
-                if rc.notes:
-                    mil.update({'notes': rc.notes})
-                mil.update({'identifier': "milestones"})
-                new_mil.append(mil)
-                pdoc = {'milestones': mil}
-            if idx > 1:
-                doc = deepcopy(all_milestones[idx - 2])
-                identifier = doc['identifier']
-                if not doc.get(
-                        'type') and not rc.type and identifier == 'milestones':
+        pdoc = {}
+        all_milestones = []
+        if rc.projectum_id and rc.milestone_uuid:
+            raise RuntimeError("A uuid fragment and projectum id have been entered.\n"
+                                "You may enter a milestone uuid or a projectum id but not both.\n"
+                                "Enter a milestone uuid to update an existing milestone, or a projectum id to add a new milestone to that projectum.\n")
+        if not rc.projectum_id and not rc.milestone_uuid:
+            raise RuntimeError("No milestone uuid or projectum id was entered.\n"
+                                "Enter a milestone uuid to update an existing milestone, or a projectum id to add a new milestone to that projectum.\n")
+
+        if rc.projectum_id:
+            key = rc.projectum_id
+            filterid = {'_id': key}
+            target_prum = rc.client.find_one(rc.database, rc.coll, filterid)
+            if not target_prum:
+                pra = fragment_retrieval(self.gtx["projecta"], ["_id"],
+                                         rc.projectum_id)
```

Comment:
Please can you check that the rc.projectum_id gets stripped somewhere.   It wasn't before so leading and trailing whitespaces would cause it to fail but they are common with copy-=paste so better behavior is that it can recover from that.

Reply:
Done.